src/EnergyPlus/IndoorGreen.hh

```diff
@@ -0,0 +1,138 @@
+    {
+        std::string IndoorGreenName;
+        std::string ZoneName;
+        std::string SurfName;
+        std::string Schedule;
+        int SchedPtr = 0;
+        int SchedLEDPtr = 0;
+        int LightRefPtr = 0; //daylight reference point number;
+        int LightControlPtr = 0; //daylight control point number;
+        int SchedLEDDaylightTargetPtr = 0; //LED-Daylight PPFD setpoint schedule pointer
+        Real64 LeafArea = 0.0;       // one-sided leaf area 
+        Real64 LEDNominalPPFD = 0.0; // nominal PPFD for LED grow light (PPFD)
+        Real64 LEDNominalEleP = 0.0; // nominal power for total LED grow light (W)
+        Real64 LEDRadFraction = 0.0; // radiant fraction of LED grow light (0-1)
+        //Real64 ZPreTemp = 0.0;
+        //Real64 ZPreHum = 0.0;
+        Real64 ZCO2 = 400;
+        Real64 ZVPD = 0.0;          // vapor pressure deficit (pa)
+        Real64 ZPPFD = 0;          //PPFD
+        Real64 SensibleRate = 0.0; //w
+        Real64 LatentRate = 0.0;   //w
+        Real64 ETRate = 0.0;       // kg/(m2s)
+        Real64 LambdaET = 0.0;     //J/(kg m2)
+        Real64 LEDActualPPFD = 0.0; // LED operational PPFD 
+        Real64 LEDActualEleP = 0.0; // LED operational power (W)
+        Real64 LEDActualEleCon = 0.0; //LED operational electricity consumption (J)
+        int SurfPtr = 0;
+        int ZoneListPtr = 0;
+        int ZonePtr = 0;                    // point to the zone where the indoor greenery system is located
+        int SpacePtr = 0;                   // point to the space where the indoor greenery system is located
+        int ETCalculationMethod = 0;        // - Method for ET calculation-(Penman-Monteith=1, Stanghellini=2, Data-driven=3)
+        int LightingMethod = 0;             // - Method for light calculation-(LED=1, Daylighting=2, LED-Daylighting=3)
+        bool CheckIndoorGreenName = true;
+        Real64 EMSET = 0.0;
+        bool EMSETCalOverrideOn = false;
+        Array1D_string FieldNames;
+     };
+
+    void SimIndoorGreen(EnergyPlusData &state);
+
+    void GetIndoorGreenInput(EnergyPlusData &state, bool &ErrorsFound);
+
+    void InitIndoorGreen(EnergyPlusData &state);
+
+    void ETModel(EnergyPlusData &state);
+
+    Real64
+    ETPenmanMonteith(EnergyPlusData &state, Real64 &ZonePreTemp, Real64 &ZonePreHum, Real64 &ZoneCO2, Real64 &ZonePPFD, Real64 &VPD, Real64 &LAI);
+    
+    Real64 ETStanghellini(EnergyPlusData &state, Real64 &ZonePreTemp, Real64 &ZonePreHum, Real64 &ZoneCO2, Real64 &ZonePPFD, Real64 &VPD, Real64 &LAI);
+
+} // namespace IndoorGreen
+
+struct IndoorGreenData : BaseGlobalStruct
+{
+    int NumIndoorGreen = 0; // Number of Indoor Greenery Systems found in input
+    bool getInputFlag = true;
+    // Object Data
+    Array1D<IndoorGreen::IndoorGreenParams> indoorgreen;  
+    void clear_state() override
+    {
+        //NumIndoorGreen = 0;
+        //indoorgreen.deallocate();
+        //getInputFlag = true;
+        *this = IndoorGreenData();
```

Comment:
A better idiom is `new (this) IndoorGreenData();`.

Reply:
Thank you! Updated.